In [18]:
import pandas as pd
import numpy as np

In [19]:
df = pd.read_csv("data.csv", low_memory=False).drop(["Unnamed: 0"], axis=1)

In [26]:
from apyori import apriori

# Выбор подходящих столбцов для анализа
df_mba = df[['НомерЗаказаНаСайте', 'ТипТовара']]
# df_mba = df_mba[:100000]

# Удаление строк с отсутствующими значениями
df_mba.dropna(inplace=True)

# Преобразование DataFrame в список транзакций
transactions = df_mba.groupby('НомерЗаказаНаСайте')['ТипТовара'].apply(list).values.tolist()

# Применение алгоритма Apriori
min_support = 0.05
min_confidence = 0.5
rules = apriori(transactions, min_support=min_support, min_confidence=min_confidence)

# Вывод результатов
for rule in rules:
    antecedent = ', '.join([str(item) for item in rule[2][0][0]])
    consequent = ', '.join([str(item) for item in rule[2][0][1]]) if rule[2][0][1] else "None"
    support = rule[1]
    confidence = rule[2][0][2]
    lift = rule[2][0][3]
    print(f"Rule: {antecedent} -> {consequent}")
    print(f"Support: {support:.4f}")
    print(f"Confidence: {confidence:.4f}")
    print(f"Lift: {lift:.4f}")
    print()

C:\Users\ermol\AppData\Local\Temp\ipykernel_18272\3200605496.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_mba.dropna(inplace=True)


Rule:  -> МГТ
Support: 0.8745
Confidence: 0.8745
Lift: 1.0000



Это правило говорит о том, что когда нет никакого предшествующего товара (антецедента), то существует высокая поддержка (0.8745) для покупки товара "МГТ" (консеквента). Поддержка указывает на то, как часто данное правило встречается в данных.

Доверие (confidence) для этого правила также составляет 0.8745, что означает, что в 87.45% случаев, когда нет предшествующего товара, покупается товар "МГТ". Доверие указывает на вероятность, что консеквент будет приобретен при условии наличия антецедента.

Lift равен 1.0000, что означает, что покупка товара "МГТ" не зависит от наличия или отсутствия других товаров. Lift показывает, насколько сильно правило связано с покупкой консеквента в сравнении с случайной покупкой. Значение 1.0000 указывает на отсутствие связи между антецедентом и консеквентом.